pkgs/applications/virtualization/docker/default.nix

```diff
@@ -11,15 +11,50 @@ with lib;
 
 stdenv.mkDerivation rec {
   name = "docker-${version}";
-  version = "1.12.3";
+  version = "1.13.0-rc3";
 
   src = fetchFromGitHub {
     owner = "docker";
     repo = "docker";
     rev = "v${version}";
-    sha256 = "0jifd35h22lgh36w1j2k97pgndjh5sppr3cwndlv0saf9618wx5k";
+    sha256 = "01g6ahks5csxmq7ir1ng8z6722arqh5spmasdcc6djjr8g1lw8yh";
   };
 
+  docker-runc = overrideDerivation runc (deriv: {
+    name = "docker-runc";
+    src = fetchFromGitHub {
+      owner = "docker";
+      repo = "runc";
+      rev = "51371867a01c467f08af739783b8beafc154c4d7";
+      sha256 = "1vyc6c59npwv8xyc1qynzz5i7kwnjgf5yda4avfnap8z4zn14m0x";
+    };
+  });
+  docker-containerd = overrideDerivation containerd (deriv: {
+    name = "docker-containerd";
+    src = fetchFromGitHub {
+      owner = "docker";
+      repo = "containerd";
+      rev = "03e5862ec0d8d3b3f750e19fca3ee367e13c090e";
+      sha256 = "184sd9dwkcba3zhxnz9grw8p81x05977p36cif2dgkhjdhv12map";
+    };
+  });
+  docker-tini = overrideDerivation tini (deriv: {
+    name = "docker-init";
+    src = fetchFromGitHub {
+      owner = "krallin";
+      repo = "tini";
+      rev = "949e6facb77383876aeff8a6944dde66b3089574";
+      sha256 = "0zj4kdis1vvc6dwn4gplqna0bs7v6d1y2zc8v80s3zi018inhznw";
+    };
+
+    patchPhase = ''
```

Comment:
In `tini` the `cmake` file is patch to remove `static` from it. But in the case of docker, `tini` (`docker-init` in the end) have to be static. This is to override the `patchPhase` — wasn't sure it was the right move to it though.